File: VNASchedule/VNASchedule/VNASchedule/Bean/BeanFAQs.cs
using SQLite;
using System;
using System.Collections.Generic;
using System.Text;

namespace VNASchedule.Bean
{
    public class BeanFAQs : BeanBase
    {
        [PrimaryKey, PrimaryKeyS]
        public int ID { get; set; }
        public string Question { get; set; }

        public string Answer { get; set; }
        public int Status { get; set; }
        public int? DepartmentId { get; set; }
        public DateTime? Created { get; set; }
        public DateTime? Modified { get; set; }

        /// <summary>
        /// </summary>
        /// <value>
        /// 1: VN
        /// 2: EN
        /// </value>
        public string Language { get; set; }
        public bool IsSelected = false;
        // <summary>
        // Lấy đường dẫn Url tương ứng lấy dữ liệu từ Server
        // </summary>
        /// <returns></returns>
        public override string GetServerUrl()
        {
            return "/ApiPublic.ashx?func=get&bname=" + this.GetType().Name;
        }
    }
}

File: lib/base/model/app/bean_faqs.dart
import 'package:floor/floor.dart';
import 'package:json_annotation/json_annotation.dart';

part 'bean_faqs.g.dart';

@JsonSerializable()
@entity
class BeanFAQs {
  @primaryKey
  @JsonKey(name: 'ID')
  late int id;

  @JsonKey(name: 'DepartmentId')
  late int departmentId;

  @JsonKey(name: 'Question')
  late String question;

  @JsonKey(name: 'Answer')
  late String answer;

  @JsonKey(name: 'Status')
  late int status;

  @JsonKey(name: 'Created')
  late String created;

  @JsonKey(name: 'Modified')
  late String modified;

  @JsonKey(name: 'Language')
  late String language;


  BeanFAQs(this.id, this.departmentId, this.question, this.answer, this.status,
      this.created, this.modified, this.language);

  factory BeanFAQs.fromJson(Map<String, dynamic> json) => _$BeanFAQsFromJson(json);

  Map<String, dynamic> toJson() => _$BeanFAQsToJson(this);
}
